wire-ios-sync-engine/Source/Use cases/AppendFileMessageUseCase.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAnalytics
+import WireDataModel
+
+public protocol AppendFileMessageUseCaseProtocol {
+    func invoke<Conversation: MessageAppendableConversation>(
+        with fileMetadata: ZMFileMetadata,
+        in conversation: Conversation
+    ) throws
+}
+
+public struct AppendFileMessageUseCase: AppendFileMessageUseCaseProtocol {
+    let analyticsSession: AnalyticsSessionProtocol?
+
+    public init(analyticsSession: AnalyticsSessionProtocol?) {
+        self.analyticsSession = analyticsSession
+    }
+
+    public func invoke<Conversation: MessageAppendableConversation>(
+        with fileMetadata: ZMFileMetadata,
+        in conversation: Conversation
+    ) throws {
+        let message = try conversation.appendFile(with: fileMetadata, nonce: UUID())
+
+        var contributionType: ContributionType = .fileMessage
+
+        if let fileMessageData = message.fileMessageData {
```

Comment:
@agisilaos You asked me offline if I had any thoughts on testing this - i.e each contribution type.

I do think it is worth testing but I would do so in tests for `ContributionType`. For example I may add an initializer to `ContributionType`:

```swift
extension ContributionType {
    init(message: ZMConversationMessage) {
        // ..
    }
}
```

Then test that implementation. `ZMConversationMessage` &  `ZMFileMessageData` are both protocols so could be mocked if you are unable to instantiate real objects.

This is just one option. I guess there are other options. It could completely be done in another PR.